python/llm/dev/benchmark/harness/bigdl_llm.py

```diff
@@ -13,116 +13,42 @@
-    return _torch_dtype
-
-class BigDLLM(BaseLM):
-    def __init__(
-        self,
-        device="xpu",
-        pretrained="gpt2",
-        revision="main",
-        low_cpu_mem_usage=None,
-        subfolder=None,
-        tokenizer=None,
-        batch_size=1,
-        load_in_8bit: Optional[bool] = False,
-        trust_remote_code: Optional[bool] = True,
-        load_in_low_bit=None,
-        dtype: Optional[Union[str, torch.dtype]] = "auto",
-        **kwargs
-    ):
-        super().__init__()
-
-        assert isinstance(pretrained, str)
-        assert isinstance(batch_size, (int,str))
-        if 'xpu' in device:
-            import intel_extension_for_pytorch as ipex
-        model = AutoModelForCausalLM.from_pretrained(pretrained,
-                                          load_in_low_bit=load_in_low_bit,
-                                          optimize_model=kwargs.get('optimize_model', True),
-                                          trust_remote_code=trust_remote_code,
-                                          use_cache=True,
-                                          torch_dtype=_get_dtype(dtype))
-        print(model) # print model to check precision
-        self._device = device
-        self.model = model.to(device)
-
-        self.tokenizer = AutoTokenizer.from_pretrained(pretrained, trust_remote_code=True)
-
-        # setup for automatic batch size detection
-        if batch_size == 'auto':
-            self.batch_size_per_gpu = batch_size
-        else:
-            self.batch_size_per_gpu = int(batch_size)
-
-    @property
-    def eot_token_id(self):
-        # we use EOT because end of *text* is more accurate for what we're doing than end of *sentence*
-        return self.tokenizer.eos_token_id
-
-    @property
-    def max_length(self):
-        return 2048  # TODO: how to get this from config
-
-    @property
-    def max_gen_toks(self):
-        return 256
-
-    @property
-    def batch_size(self):
-        # TODO: fix multi-gpu
-        return self.batch_size_per_gpu  # * gpus
+from bigdl.llm.transformers import AutoModelForCausalLM
+
+import inspect
+from lm_eval.models.huggingface import AutoCausalLM
+from lm_eval import utils
+from functools import partial
+
+
+class BigDLLM(AutoCausalLM):
+    AUTO_MODEL_CLASS = AutoModelForCausalLM
+    AutoCausalLM_ARGS = inspect.getfullargspec(AutoCausalLM.__init__).args
+    def __init__(self, *args, **kwargs):
+        # wrap  and force the Reorderer to be in a decrease order
+        # This is a workaround to avoid frequent memory allocation which may cause OOM
+        def force_decrease_order(Reorderer):
+            def DecreaseReorderer(arr, fn):
+                def _collate(x):
+                    len, tokens = fn(x)
+                    len = - abs(len)
+                    return len, tokens
+                return Reorderer(arr, _collate)
+            return DecreaseReorderer
+        utils.Reorderer = force_decrease_order(utils.Reorderer)
```

Comment:
Don't need to put this in the class init?
Just put them in the global range of this file?